tests/common/snappi/common_helpers.py

```diff
@@ -436,68 +448,113 @@ def config_wred(host_ans, kmin, kmax, pmax, profile=None):
+
     Args:
         host_ans: Ansible host instance of the device
         prio (int): priority
+        asic_value: asic value of the host
+
     Returns:
         N/A
     """
-    host_ans.shell('sudo ecnconfig -q {} on'.format(prio))
+    if asic_value == 'None':
+        host_ans.shell('sudo ecnconfig -q {} on'.format(prio))
+    else:
+        host_ans.shell('sudo ip netns exec {} ecnconfig -q {} on'.format(asic_value, prio))
 
 
-def disable_ecn(host_ans, prio):
+def disable_ecn(host_ans, prio, asic_value='None'):
     """
     Disable ECN marking on a priority
+
     Args:
         host_ans: Ansible host instance of the device
         prio (int): priority
+        asic_value: asic value of the host
+
     Returns:
         N/A
     """
-    host_ans.shell('sudo ecnconfig -q {} off'.format(prio))
+    if asic_value == 'None':
+        host_ans.shell('sudo ecnconfig -q {} off'.format(prio))
+    else:
+        asic_type = str(host_ans.facts["asic_type"])
+        host_ans.shell('sudo ip netns exec {} ecnconfig -q {} off'.format(asic_value, prio))
+        if asic_type == 'broadcom':
+            try:
+                host_ans.shell('bcmcmd -n {} "BCMSAI credit-watchdog enable"'.format(asic_value))
+            except Exception:
+                host_ans.shell('bcmcmd -n {} "BCMSAI credit-watchdog enable"'.format(asic_value[-1]))
 
 
-def config_buffer_alpha(host_ans, profile, alpha_log2):
+def config_buffer_alpha(host_ans, profile, alpha_log2, asic_value='None'):
     """
     Configure buffer threshold (a.k.a., alpha)
+
     Args:
         host_ans: Ansible host instance of the device
         profile (str): buffer profile name
         alpha_log2 (int): set threshold to 2^alpha_log2
+        asic_value: asic value of the host
+
     Returns:
         N/A
     """
-    host_ans.shell('sudo mmuconfig -p {} -a {}'.format(profile, alpha_log2))
+    if asic_value == 'None':
+        host_ans.shell('sudo mmuconfig -p {} -a {}'.format(profile, alpha_log2))
+    else:
+        host_ans.shell('sudo ip netns exec {} mmuconfig -p {} -a {}'.format(asic_value, profile, alpha_log2))
 
 
-def config_ingress_lossless_buffer_alpha(host_ans, alpha_log2):
+def config_ingress_lossless_buffer_alpha(host_ans, alpha_log2, asic_value='None'):
     """
     Configure ingress buffer thresholds (a.k.a., alpha) of a device to 2^alpha_log2
+
     Args:
         host_ans: Ansible host instance of the device
         alpha_log2 (int): set threshold to 2^alpha_log2
+        asic_value: asic value of the host
+
     Returns:
         If configuration succeeds (bool)
     """
     if not isinstance(alpha_log2, int):
         return False
 
-    config_facts = host_ans.config_facts(host=host_ans.hostname,
-                                         source="running")['ansible_facts']
+    if asic_value == 'None':
+        config_facts = host_ans.config_facts(host=host_ans.hostname, source="running")['ansible_facts']
+    else:
+        config_facts = host_ans.config_facts(
```

Comment:
Why can't namespace be set to None in config_facts for single ASIC systems?

Reply:
Since this is based on users choice of selection of ports, while creating the port configuration we are appending the asic value if the setup is multi-chassis based or 'None' string if its a non chassis based dut, so that the script selects the ports accordingly from the ansible files executes